src/BizHawk.WinForms.Controls/LabelEx/HexLabelEx.cs

```diff
@@ -0,0 +1,128 @@
+using System.ComponentModel;
+using System.Drawing;
+using System.Windows.Forms;
+using BizHawk.Common;
+
+
+namespace BizHawk.WinForms.Controls
+{
+	/// <inheritdoc cref="Docs.LabelOrLinkLabel"/>
+	public class HexLabelEx : LabelExBase
+	{
+		[DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+		public new bool AutoSize => base.AutoSize;
+
+		[DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+		public new Size Size => base.Size;
+
+		public Color ZeroColor { get; set; } = Color.SlateGray;
+
+		private float spacingHexModifier;
+		private float spacingPreDivider;
+		private float spacingPostDividerModifier;
+		private float spacingAscii;
+		private float spacingLineModifier;
+
+		public HexLabelEx()
+		{
+			base.AutoSize = true;
+			this.BackColor = Color.Transparent;
+
+			spacingHexModifier = 0.7F;
+			spacingPreDivider = 3.0F;
+			spacingPostDividerModifier = 3.5F;
+			spacingAscii = 7.0F;
+			spacingLineModifier = 0.56F;
+
+			if (OSTailoredCode.IsUnixHost)
+			{
+				// TODO: spacing values will probably be different on linux
+			}
+		}
+
+		protected override void OnPaint(PaintEventArgs e)
+		{
+			string text = this.Text;
+			Font font = this.Font;
+			PointF point = new PointF(0, 0);
+			char gap = ' ';
+
+			string[] lines = text.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+
+			Color color = this.ForeColor;
+
+			foreach (string line in lines)
+			{
+				// split left and right panes
+				string[] panes = line.Split('|');
```

Comment:
Really don't like this. The caller has full knowledge of the content that's being passed in and should be able to make separate calls before or instead of passing in a concatenated line.